compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -534,4 +518,82 @@ private void validateReturnParameter(DocumentableNode documentableNode, BLangNod
             returnParameter.setReturnType(((BLangFunction) node).getReturnTypeNode().type);
         }
     }
+
+    // List that holds fields that are documented at field level.
+    private List<String> getDocumentedFields(List<? extends SimpleVariableNode> actualParameters) {
+        List<String> fieldsDocumentedAtFieldLevel = new ArrayList<>();
+        for (SimpleVariableNode field : actualParameters) {
+            if (field.getMarkdownDocumentationAttachment() == null) {
+                continue;
+            }
+            fieldsDocumentedAtFieldLevel.add(field.getName().getValue());
+        }
+        return fieldsDocumentedAtFieldLevel;
+    }
+
+    private Map<String, BLangMarkdownParameterDocumentation> getDocumentedParameters(
+            LinkedList<BLangMarkdownParameterDocumentation> deprecatedParameters,
+            List<String> fieldsDocumentedFields,
+            DiagnosticCode parameterAlreadyDefined) {
+        Map<String, BLangMarkdownParameterDocumentation> documentedDeprecatedParameterMap = new HashMap<>();
+
+        for (BLangMarkdownParameterDocumentation parameter : deprecatedParameters) {
+            String parameterName = parameter.getParameterName().getValue();
+            // Check for parameters which are documented multiple times.
+            if (documentedDeprecatedParameterMap.containsKey(parameterName) ||
+                    fieldsDocumentedFields.contains(parameterName)) {
+                dlog.warning(parameter.pos, parameterAlreadyDefined, parameterName);
+                continue;
+            }
+            documentedDeprecatedParameterMap.put(parameterName, parameter);
+        }
+        return documentedDeprecatedParameterMap;
+    }
+
+    private void validateDeprecatedParameters(DocumentableNode documentableNode,
+                                             List<BLangSimpleVariable> actualParameters,
+                                             BLangSimpleVariable restParam,
+                                             DiagnosticCode parameterAlreadyDefined,
+                                             DiagnosticCode noSuchParameter) {
+        BLangMarkdownDocumentation documentation = documentableNode.getMarkdownDocumentationAttachment();
+        if (documentation == null) {
+            return;
+        }
+
+        List<String> fieldsDocumentedAtFieldLevel = getDocumentedFields(actualParameters);
+
+        Map<String, BLangMarkdownParameterDocumentation> documentedDeprecatedParameterMap = new HashMap<>();
+        if (documentation.deprecatedParametersDocumentation != null) {
```

Comment:
Can't we just return here if `documentation.deprecatedParametersDocumentation != null`? Will remove the nesting and the initialization to empty map at L565.

Reply:
When the markdown doc is available and deprecation doc is not available(an error scenario), we need to pass this empty map to compare with actual params. See L573